src/main/java/mate/academy/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+package mate.academy.bookshop.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.order.OrderRequestDto;
+import mate.academy.bookshop.dto.order.OrderResponseDto;
+import mate.academy.bookshop.dto.order.UpdateOrderRequestDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.OrderMapper;
+import mate.academy.bookshop.model.CartItem;
+import mate.academy.bookshop.model.Order;
+import mate.academy.bookshop.model.OrderItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.model.User;
+import mate.academy.bookshop.repository.CartItemRepository;
+import mate.academy.bookshop.repository.OrderItemRepository;
+import mate.academy.bookshop.repository.OrderRepository;
+import mate.academy.bookshop.repository.ShoppingCartRepository;
+import mate.academy.bookshop.service.OrderService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeAnOrder(Long id, OrderRequestDto requestDto) {
```

Comment:
```suggestion
    public OrderResponseDto placeOrder(Long id, OrderRequestDto requestDto) {
```
articles are usually not included in functions' names